pycloudlib/lxd/instance.py

```diff
@@ -81,21 +81,37 @@ def ip(self):
         Returns:
             IP address assigned to instance.
 
+        Raises: TimeoutError when exhausting retries trying to parse lxc list
+            for ip addresses.
         """
         retries = 150
 
         while retries != 0:
-            command = 'lxc list {} -c 4 --format csv'.format(self.name)
-            result = subp(command.split()).stdout
-
-            if result != '':
-                break
-
+            command = [
+                 'lxc', 'list', '^{}$'.format(self.name), '-c4',
+                 '--format', 'csv'
+            ]
+            result = subp(command)
+            if result.ok and result.stdout:
+                ip_address = None
+                try:
+                    # Expect "<ip> (ethX)" when network fully configured
+                    ip_address, _dev = result.stdout.split()
```

Comment:
I don't believe I've seen a case where this split has failed: do we have a motivating error case for this?